bindings/python/iota_sdk/types/payload.py

```diff
@@ -3,50 +3,88 @@
 
 from __future__ import annotations
 from iota_sdk.types.common import HexStr
+from iota_sdk.types.output import Output
+from iota_sdk.types.input import UtxoInput
 from iota_sdk.types.signature import Ed25519Signature
+from iota_sdk.types.unlock import Unlock
+from dacite import from_dict
 from dataclasses import dataclass
-from enum import Enum
+from enum import IntEnum
 from typing import Any, Optional, List
 
 
-class PayloadType(Enum):
+class PayloadType(IntEnum):
     TreasuryTransaction = 4
     TaggedData = 5
     Transaction = 6
     Milestone = 7
 
 
-class Payload():
-    def __init__(self, type, milestone: Optional[Any] = None, tagged_data=None,
-                 transaction=None, treasury_transaction: Optional[Any] = None):
-        """Initialize a payload
-        """
-        self.type = type
-        self.milestone = milestone
-        self.tagged_data = tagged_data
-        self.transaction = transaction
-        self.treasury_transaction = treasury_transaction
+@dataclass
+class RegularTransactionEssence:
+    type: int
+    networkId: str
+    inputsCommitment: HexStr
+    inputs: List[UtxoInput]
+    outputs: List[Output]
+    payload: Optional[Payload] = None
 
     def as_dict(self):
         config = {k: v for k, v in self.__dict__.items() if v is not None}
 
-        if "milestone" in config:
-            del config["milestone"]
-        if "tagged_data" in config:
-            del config["tagged_data"]
-        if "transaction" in config:
-            del config["transaction"]
-        if "treasury_transaction" in config:
-            del config["treasury_transaction"]
+        if 'payload' in config:
+            config['payload'] = config['payload'].as_dict()
+
+        config['inputs'] = list(map(
+            lambda x: x.__dict__, config['inputs']))
+
+        config['outputs'] = list(map(
+            lambda x: x.as_dict(), config['outputs']))
+
+        return config
+
+
+@dataclass
+class Payload():
+    """Initialize a Payload.
+    """
+    type: int
+    # tagged data
+    tag: Optional[HexStr] = None
+    data: Optional[HexStr] = None
+    # transaction
+    essence: Optional[RegularTransactionEssence] = None
+    unlocks: Optional[List[Unlock]] = None
+    # milestone
```

Comment:
But milestone also had an essence field? How does that work?

Reply:
There is no essence field in the dto

Reply:
Oh ok